subsys/zbus/zbus.c

```diff
@@ -83,15 +152,20 @@ static inline int _zbus_vded_exec(const struct zbus_channel *chan, k_timepoint_t
 			continue;
 		}
 
-		err = _zbus_notify_observer(chan, obs, end_time);
-
-		_ZBUS_ASSERT(err == 0,
-			     "could not deliver notification to observer %s. Error code %d",
-			     _ZBUS_OBS_NAME(obs), err);
+		err = _zbus_notify_observer(chan, obs, end_time, buf);
 
 		if (err) {
 			last_error = err;
+			LOG_ERR("could not deliver notification to observer %s. Error code %d",
+				_ZBUS_OBS_NAME(obs), err);
+			if (err == -ENOMEM) {
+				IF_ENABLED(CONFIG_ZBUS_MSG_SUBSCRIBER, (net_buf_unref(buf);))
+				return err;
+			}
 		}
+
+		IF_ENABLED(CONFIG_ZBUS_LOG_LEVEL_DBG,
+			   (LOG_DBG(" %d -> %s", index++, _ZBUS_OBS_NAME(obs)));)
```

Comment:
just `LOG_DBG`? should already be a no-op if the log level is low enough

Reply:
No, because of the index variable. 
```c
137	IF_ENABLED(CONFIG_ZBUS_LOG_LEVEL_DBG, (int index = 0;))
```

Reply:
Oh I see, I think it would be more readable if you leave the index there all the time, flag it as `__maybe_unused` to keep the compiler happy and just use the debug print normally

Reply:
Great Idea! Done.